app/controllers/events_controller.rb

```diff
@@ -0,0 +1,40 @@
+class EventsController < ApplicationController
+  before_action :authenticate_user!, except: [:index]
+
+  def index
+    @events = Event.all
+  end
+
+  def show
+    @event = Event.find(params[:id])
+    @attendees = @event.attendees
+  end
+
+  def new
+    @event = Event.new
+  end
+
+  def create
+    @event = current_user.created_events.build(event_params)
+    @event.save
+    if @event.save
+      redirect_to @event, notice: 'Your event was created'
+    else
+      render :new
+    end
+  end
```

Comment:
when we create an event, we need to validate it first and then create it, calling twice the save action is not a good practice, and can cause some double creation if there isn't proper validation to it, so we can remove the first call, or do it the following way (best practice)
```ruby
 def create
    @event = current_user.created_events.build(event_params)
    if @event.valid?
      @event.save
      redirect_to @event, notice: 'Your event was created'
    else
      render :new
    end
  end
```